### PR TITLE
add ticket 7 POST comment

### PR DESCRIPTION
**Pull request by abrasumente999**
- plus refactor util function checkExists to be ambiguous and reusable

### CODE REVIEW

File: controllers/reviews.controller.js [hunk 1]
```diff
@@ -40,3 +42,15 @@ exports.getCommentsByReviewId = (req, res, next) => {
...
+  try {
+    const commentObj = req.body;
+    const review_id = req.params;
+
+    const comment = await insertComment(review_id, commentObj);
+    res.status(201).send({ comment });
+  } catch (err) {
+    next(err);
+  }
+};
```
Comment (emilyb93) [inline controllers/reviews.controller.js]: If you're going to go with async/await, make sure you refactor all your functions at some point but ill let it slide for now

File: models/reviews.model.js [hunk 2]
```diff
@@ -76,13 +76,34 @@ exports.selectReviewsById = (review_id) => {
...
+};
+
+exports.insertComment = async (id, comment) => {
+  const { review_id } = id;
+  const { username, body } = comment;
+
+  const queryStr = `INSERT INTO comments(author, body, review_id)
+  SELECT $1, $2, (SELECT review_id FROM reviews WHERE review_id = $3::INT)
+  FROM comments
+  RETURNING*;`;
```
Comment (emilyb93) [inline models/reviews.model.js]: This select is giving you a false positive here. Its not actually inserting the values as far as i know. You'll want this to look a bit more like
INSERT INTO comments(x,y,z) VALUES($1, $2, $3) RETURNING*;

**Review by emilyb93 — CHANGES_REQUESTED** [review]
Just want you to redo that query please

**Comment by abrasumente999** [comment]
Done, also managed to refactor without the async (as it wasnt passing earlier without it for some reason)!

**Review by emilyb93 — APPROVED** [review]

**Pull request merged**